### PR TITLE
Update stream llm to get correct outputs and re-enable rerotated-attention test.

### PR DESCRIPTION
**Pull request by raikonenfnu**
During the update of pytorch/HF, there seem to be a change of how causal mask was being handled. It seems like the attention.forward function used to get a `causal_mask` from the argument as `attention_mask` when is_causal is on. Now it seems like we would need to construct our own mask when `is_causal` is true. This was causing numerical issues in this test as well as on Llama2 qualitatively. 

This PR introduces construction of causal mask, as well as removing unnecessary tensor parallel config checks which simplifies the code quite a bit.

### CODE REVIEW

**Review by monorimet — APPROVED** [review]
Nice, thanks for this fix! LGTM
